governance/program/src/state.rs

```diff
@@ -0,0 +1,281 @@
+//! Program accounts
+
+use solana_program::{epoch_schedule::Slot, instruction::Instruction, pubkey::Pubkey};
+
+/// Max number of instructions allowed for a proposal
+pub const MAX_INSTRUCTIONS: usize = 5;
+
+/// Defines all Governance accounts types
+#[derive(Clone, Debug, PartialEq)]
+pub enum GovernanceAccountType {
+    /// Default uninitialized account state
+    Uninitialized,
+
+    /// Top level aggregation for governances within Governance Token (and optional Council Token).
+    Realm,
+
+    /// Voter record for each voter within a Realm.
+    VoterRecord,
+
+    /// Program Governance account.
+    ProgramGovernance,
+
+    /// Proposal account for Governance account. A single Governance account can have multiple Proposal accounts.
+    Proposal,
+
+    /// Proposal voting state account. Every Proposal account has exactly one ProposalState account.
+    ProposalState,
+
+    /// Vote record account for a given Proposal.  Proposal can have 0..n voting records.
+    ProposalVoteRecord,
+
+    /// Single Signer Instruction account which holds an instruction to execute for Proposal.
+    SingleSignerInstruction,
+}
+
+impl Default for GovernanceAccountType {
+    fn default() -> Self {
+        GovernanceAccountType::Uninitialized
+    }
+}
+
+/// Vote  with number of votes
+#[derive(Clone, Debug, PartialEq)]
+pub enum VoteWeight {
+    /// Yes vote
+    Yes(u64),
+
+    /// No vote
+    No(u64),
+}
+
+/// Yes/No Vote
+#[derive(Clone, Debug, PartialEq)]
+pub enum Vote {
+    /// Yes vote
+    Yes,
+    /// No vote
+    No,
+}
+
+/// Governance Realm Account
+/// Account PDA seeds" ['governance', name]
+pub struct Realm {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Governance mint
+    pub governance_mint: Pubkey,
```

Comment:
nit: I feel the Governance mint (and all associated naming around it) is slightly confusing, since the entire program is also called Governance.

The Council is a smaller group, so perhaps we can come up with a different term than "Governance" for the entire population of users, some ideas:
* User mint
* Everybody mint
* Population mint
* Masses mint
Not liking any of these so much.  But what do you think?

Reply:
"User mint" actually doesn't seem so bad.   eg, the users can vote on a proposal or the council can.

Reply:
Yeah, I also thought it was a bit confusing initially but couldn't find a good alternative. 
'Users mint' seems a bit to vague to me.
But I think I have a good one now. I renamed it to Community Token / Mint. This way we can have either the general large community voting or a smaller council.